modules.d/10i18n/module-setup.sh

```diff
@@ -216,7 +217,17 @@ install() {
         done
 
         for keymap in "${!KEYMAPS[@]}"; do
-            inst_opt_decompress "${keymap}"
+            if [[ -L ${keymap} ]]; then
+                maplink=$(readlink -f "${keymap}")
+                # skip symlinked directories
+                [[ -d ${maplink} ]] && continue
+
+                inst_opt_decompress "${maplink}"
+                # create new symlink to decompressed keymap
+                ln -srn "${initdir}${maplink%.gz}" "${initdir}${keymap%.gz}"
```

Comment:
Thanks to #2618, I found an issue in this `ln` with `--sysroot`:

```
+ for keymap in "${!KEYMAPS[@]}"
+ [[ -L /.snapshots/20/snapshot/usr/share/kbd/keymaps/xkb/kr.map.gz ]]
++ readlink -f /.snapshots/20/snapshot/usr/share/kbd/keymaps/xkb/kr.map.gz
+ maplink=/.snapshots/20/snapshot/usr/share/kbd/keymaps/xkb/us.map.gz
+ [[ -d /.snapshots/20/snapshot/usr/share/kbd/keymaps/xkb/us.map.gz ]]
+ inst_opt_decompress /.snapshots/20/snapshot/usr/share/kbd/keymaps/xkb/us.map.gz
+ ln -srn /var/tmp/dracut.PDue2y/initramfs/.snapshots/20/snapshot/usr/share/kbd/keymaps/xkb/us.map /var/tmp/dracut.PDue2y/initramfs/.snapshots/20/snapshot/usr/share/kbd/keymaps/xkb/kr.map
ln: failed to create symbolic link '/var/tmp/dracut.PDue2y/initramfs/.snapshots/20/snapshot/usr/share/kbd/keymaps/xkb/kr.map': No such file or directory
```

So, `$dracutsysrootdir` must be removed:

```diff
- ln -srn "${initdir}${maplink%.gz}" "${initdir}${keymap%.gz}"
+ maplink=${maplink%.gz}
+ keymap=${keymap%.gz}
+ ln -srn "${initdir}${maplink#"$dracutsysrootdir"}" "${initdir}${keymap#"$dracutsysrootdir"}"
```

Reply:
Thanks for the hint.
...done